rskj-core/src/main/java/org/ethereum/rpc/JsonRpcWeb3ServerHandler.java

```diff
@@ -0,0 +1,123 @@
+package org.ethereum.rpc;
+
+import co.rsk.rpc.CorsConfiguration;
+import co.rsk.rpc.JsonRpcFilterServer;
+import co.rsk.rpc.ModuleDescription;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.googlecode.jsonrpc4j.AnnotationsErrorResolver;
+import com.googlecode.jsonrpc4j.DefaultErrorResolver;
+import com.googlecode.jsonrpc4j.MultipleErrorResolver;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.*;
+import org.ethereum.rpc.exception.RskErrorResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ChannelHandler.Sharable
+public class JsonRpcWeb3ServerHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("jsonrpc");
+    private static final int JSON_RPC_PARSE_ERROR_CODE = -32700;
+    private static final int JSON_RPC_INVALID_PARAMS_CODE = -32602;
+    private static final int JSON_RPC_INTERNAL_ERROR_CODE = -32603;
+    private static final int JSON_RPC_SERVER_ERROR_LOW_CODE = -32000;
+    private static final int JSON_RPC_SERVER_ERROR_HIGH_CODE = -32099;
+    private static final int JSON_RPC_INVALID_REQUEST_CODE = -32600;
+    private static final int JSON_RPC_METHOD_NOT_FOUND_CODE = -32601;
+
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final JsonNodeFactory jsonNodeFactory = JsonNodeFactory.instance;
+    private final JsonRpcFilterServer jsonRpcServer;
+    private final CorsConfiguration corsConfiguration;
+
+    public JsonRpcWeb3ServerHandler(Web3 service, List<ModuleDescription> filteredModules, CorsConfiguration corsConfiguration) {
+        this.corsConfiguration = corsConfiguration;
+        this.jsonRpcServer = new JsonRpcFilterServer(service, service.getClass(), filteredModules);
+        jsonRpcServer.setErrorResolver(new MultipleErrorResolver(new RskErrorResolver(), AnnotationsErrorResolver.INSTANCE, DefaultErrorResolver.INSTANCE));
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) throws Exception {
+        HttpMethod httpMethod = request.getMethod();
+        HttpResponse response;
+        if (HttpMethod.OPTIONS.equals(httpMethod)) {
```

Comment:
Maybe you could implement a sort of Chain Of Responsibility to handle the different possible methods of the http Request and get the response like that, instead of having this if/else/if/else structure